run.py

```diff
@@ -79,57 +76,42 @@ def get_watson_online_store():
+        cloudant_db_name = os.environ.get(
+            "CLOUDANT_DB_NAME") or 'watson_online_store'
         discovery_username = os.environ.get('DISCOVERY_USERNAME')
         discovery_password = os.environ.get('DISCOVERY_PASSWORD')
         discovery_url = os.environ.get('DISCOVERY_URL')
 
-        if not all((conversation_username,
-                    conversation_password,
-                    cloudant_username,
-                    cloudant_password,
-                    cloudant_url)):
-            # If some of the service env vars are not set get them from VCAP
-            vcap_env = None
+        # If the CLOUDANT_USERNAME env var was not set then use
+        # VCAP_SERVICES like a WatsonService would.
+        if not cloudant_username:
             vcap_services = os.environ.get("VCAP_SERVICES")
-            if vcap_services:
-                vcap_env = json.loads(vcap_services)
+            vcap_env = json.loads(vcap_services) if vcap_services else None
             if vcap_env:
-                conversation_creds = WatsonEnv.get_vcap_credentials(
-                    vcap_env, 'conversation')
-                conversation_username = \
-                    conversation_username or conversation_creds['username']
-                conversation_password = \
-                    conversation_password or conversation_creds['password']
-
                 cloudant_creds = WatsonEnv.get_vcap_credentials(
                     vcap_env, 'cloudantNoSQLDB')
-                cloudant_username = \
-                    cloudant_username or cloudant_creds['username']
-                cloudant_password = \
-                    cloudant_password or cloudant_creds['password']
-                cloudant_url = cloudant_url or cloudant_creds['url']
-
-        # If we still don't have all the above plus a few, then no WOS.
-        if not all((slack_bot_token,
-                    conversation_username,
-                    conversation_password,
-                    cloudant_username,
-                    cloudant_password,
-                    cloudant_url,
-                    cloudant_db_name)):
-
-            print(MISSING_ENV_VARS)
-            return None
+                if cloudant_creds:
+                    cloudant_url = cloudant_creds['url']  # overrides default
+                    if 'username' in cloudant_creds:
+                        cloudant_username = cloudant_creds['username']
+                    if 'password' in cloudant_creds:
+                        cloudant_password = cloudant_creds['password']
 
         # Instantiate Watson Assistant client.
-        conversation_client = ConversationV1(
-            username=conversation_username,
-            password=conversation_password,
-            version='2017-05-26')
+        # - only give a url if we have one (don't override the default)
+        conversation_kwargs = {
+            'version': '2017-05-26',
+            'username': conversation_username,
```

Comment:
isn't `conversation_username` and the other waston bits only grabbed from `conversation_username = os.environ.get("CONVERSATION_USERNAME")` ? 

how are we reading from VCAP services?

Reply:
The SDK already has code to read from VCAP so we just let the SDK do its thing instead of writing our own slightly different way of doing it.  Cloudant unfortunately is not part of the watson-developer-sdk so we still have vcap code for that part and the logic is now consistent with the SDK inits.

Reply:
FYI -- python SDK WatsonService getting VCAP if needed here:

https://github.com/watson-developer-cloud/python-sdk/blob/develop/watson_developer_cloud/watson_service.py#L179